### DISCUSSION TITLE
Sign up with existing email spec not working (and potential fix)

### DISCUSSION THREAD

**Discussion by pavgup** [fastapi/full-stack-fastapi-template#1400] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
The current spec has this code:

test("Sign up with existing email", async ({ page }) => {
  const fullName = "Test User"
  const email = randomEmail()
  const password = randomPassword()

  // Sign up with an email
  await page.goto("/signup")

  await fillForm(page, fullName, email, password, password)
  await page.getByRole("button", { name: "Sign Up" }).click()

  // Sign up again with the same email
  await page.goto("/signup")

  await fillForm(page, fullName, email, password, password)
  await page.getByRole("button", { name: "Sign Up" }).click()

  await page
    .getByText("The user with this email already exists in the system")
    .click()
})
```

### Description

When the spec to test signing up with the same email address, the spec fails with an interesting problem:

1. The backend produces a 500 error because the database is unable to insert the ix_user_email that is not unique. It means the backend, when it searched for the users to detect a duplicate, was not able to find this user somehow.
2. This causes the spec to fail with a CORS error. This is super confusing, but after doing enough research, it's clear that 500 errors from fastapi result in CORS errors because they don't take the cors middleware configurations from the app.
3. And as I played with this, it became clear that the spec might just be moving too quickly for the backend to process --

So, a potential fix might look like this:

```
test("Sign up with existing email", async ({ page }) => {
  const fullName = "Test User"
  const email = randomEmail()
  const password = randomPassword()

  // Sign up with an email
  await page.goto("/signup")

  await fillForm(page, fullName, email, password, password)
  await page.getByRole("button", { name: "Sign Up" }).click()

  await page
    .getByText("Your account has been created successfully")
    .click()
    
  // Sign up again with the same email
  await page.goto("/signup")

  await fillForm(page, fullName, email, password, password)
  await page.getByRole("button", { name: "Sign Up" }).click()

  await page
    .getByText("The user with this email already exists in the system")
    .click()
})```

### Operating System

macOS

### Operating System Details

M2 pro, sequoia 15.0.1

### Python Version

Python 3.11.10

### Additional Context

_No response_

**Comment by YuriiMotov** [comment]
Can't reproduce this problem.
Do you mean this happens sometimes, not every time?
I ran tests 3 times and did not get any `500` errors
